exporter/azureblobexporter/marshaller.go

```diff
@@ -0,0 +1,112 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package azureblobexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azureblobexporter"
+
+import (
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// Marshaler configuration used for marhsaling Protobuf
+var tracesMarshalers = map[string]ptrace.Marshaler{
+	formatTypeJSON:  &ptrace.JSONMarshaler{},
+	formatTypeProto: &ptrace.ProtoMarshaler{},
+}
+
+var metricsMarshalers = map[string]pmetric.Marshaler{
+	formatTypeJSON:  &pmetric.JSONMarshaler{},
+	formatTypeProto: &pmetric.ProtoMarshaler{},
+}
+
+var logsMarshalers = map[string]plog.Marshaler{
+	formatTypeJSON:  &plog.JSONMarshaler{},
+	formatTypeProto: &plog.ProtoMarshaler{},
+}
+
+type marshaller struct {
+	tracesMarshaler  ptrace.Marshaler
+	metricsMarshaler pmetric.Marshaler
+	logsMarshaler    plog.Marshaler
+}
+
+func newMarshaller(config *Config, host component.Host) (*marshaller, error) {
+	if config.FormatType != formatTypeJSON && config.FormatType != formatTypeProto {
+		return nil, fmt.Errorf("unsupported format type %q", config.FormatType)
+	}
+	marshaller := &marshaller{
+		tracesMarshaler:  tracesMarshalers[config.FormatType],
+		metricsMarshaler: metricsMarshalers[config.FormatType],
+		logsMarshaler:    logsMarshalers[config.FormatType],
+	}
+
+	if config.Encodings.Logs != nil {
+		encoding := host.GetExtensions()[*config.Encodings.Logs]
+		if encoding == nil {
+			return nil, fmt.Errorf("unknown encoding %q", config.Encodings.Logs)
+		}
+		// cast with ok to avoid panics.
```

Comment:
If we're casting to avoid panics, we should check the ok and return an error.

Reply:
updated